webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/ncf/TechniqueWriter.scala

```diff
@@ -149,6 +151,14 @@ class TechniqueWriter (
                      }
 
       _          <- archiver.deleteTechnique(techniqueName,techniqueVersion, category.id.name.value, modId,committer, s"Deleting technique ${techniqueName}/${techniqueVersion}")
```

Comment:
You need to first delete active technique, then on FS, because active technique deletion won't work if there's directives

Reply:
we prevent it if there is directives, or if we force the deletion we delete directives first, but I'm ok with you we must do this before